hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OmTableInsightTask.java

```diff
@@ -0,0 +1,360 @@
+   * @return A Triple with three Long values representing the count,
+   *         unreplicated size and replicated size.
+   * @throws IOException If an I/O error occurs during the iterator traversal.
+   */
+  private Triple<Long, Long, Long> getTableSizeAndCount(
+      TableIterator<String, ? extends Table.KeyValue<String, ?>> iterator)
+      throws IOException {
+    long count = 0;
+    long unReplicatedSize = 0;
+    long replicatedSize = 0;
+
+    if (iterator != null) {
+      while (iterator.hasNext()) {
+        Table.KeyValue<String, ?> kv = iterator.next();
+        if (kv != null && kv.getValue() != null) {
+          if (kv.getValue() instanceof OmKeyInfo) {
+            OmKeyInfo omKeyInfo = (OmKeyInfo) kv.getValue();
+            unReplicatedSize += omKeyInfo.getDataSize();
+            replicatedSize += omKeyInfo.getReplicatedSize();
+          }
+          count++;  // Increment count for each row
+        }
+      }
+    }
+
+    return Triple.of(count, unReplicatedSize, replicatedSize);
+  }
+
+  /**
+   * Returns a collection of table names that require data size calculation.
+   */
+  public Collection<String> getTablesToCalculateSize() {
+    List<String> taskTables = new ArrayList<>();
+    taskTables.add(OPEN_KEY_TABLE);
+    taskTables.add(OPEN_FILE_TABLE);
+    return taskTables;
+  }
+
+  @Override
+  public String getTaskName() {
+    return "OmTableInsightTask";
+  }
+
+  public Collection<String> getTaskTables() {
+    return new ArrayList<>(reconOMMetadataManager.listTableNames());
+  }
+
+  /**
+   * Read the update events and update the count and sizes of respective object
+   * (volume, bucket, key etc.) based on the action (put or delete).
+   *
+   * @param events Update events - PUT, DELETE and UPDATE.
+   * @return Pair
+   */
+  @Override
+  public Pair<String, Boolean> process(OMUpdateEventBatch events) {
+    Iterator<OMDBUpdateEvent> eventIterator = events.getIterator();
+    HashMap<String, Long> objectCountMap = initializeCountMap();
+    HashMap<String, Long> unreplicatedSizeCountMap = initializeSizeMap(false);
+    HashMap<String, Long> replicatedSizeCountMap = initializeSizeMap(true);
+    final Collection<String> taskTables = getTaskTables();
+    final Collection<String> sizeRelatedTables = getTablesToCalculateSize();
+
+    while (eventIterator.hasNext()) {
+      OMDBUpdateEvent<String, Object> omdbUpdateEvent = eventIterator.next();
+      String tableName = omdbUpdateEvent.getTable();
+
+      if (!taskTables.contains(tableName)) {
+        continue;
+      }
+
+      String countKey = getTableCountKeyFromTable(tableName);
+      String unReplicatedSizeKey =
+          getUnReplicatedSizeKeyFromTable(tableName);
+      String replicatedSizeKey =
+          getReplicatedSizeKeyFromTable(tableName);
+
+      try {
+        switch (omdbUpdateEvent.getAction()) {
+        case PUT:
+          objectCountMap.computeIfPresent(countKey, (k, count) -> count + 1L);
```

Comment:
How about case where same open key/file is updated for case, continuous update with new blocks data, and also HSync case where present in both open key/file and normal key/file table? it will update with differential data or keep added data.

Reply:
Thanks a lot for the find @sumitagrawl !
Yes it's something that I might have missed out on so here is my understanding for the following, please correct me if I am missing anything out :- 

The issue you described is related to the handling of size updates for a key in the put method. Let's break it down step by step:

1. Initially, a put event is received for a key called "file1" with a size of 10MB. This event triggers the addition of the key and its size to the unreplicatedSizeCountMap and replicatedSizeCountMap. The map keys are table names, and the values represent the total size added using update events.

2. Now, let's consider a subsequent put event for the same key "file1" where the size is updated to 15MB. In this case, the issue arises because the current implementation adds the updated size directly to the size count maps instead of considering the difference in sizes since its the same key.

As a result, when the size update event is processed, the code adds the updated size of 15MB to the existing value in the size count maps. This leads to an incorrect representation of the total size. Instead of reflecting the incremental increase of 5MB, the map value becomes 15MB, which is the updated size.

Reply:
Thanks for confirming @sumitagrawl I'll add a UPDATE event to the task!